vital/types/track.cxx

```diff
@@ -187,11 +187,42 @@ ::remove( track_state_sptr const& state )
     return false;
   }
 
-  (*pos)->track_.reset();
-  this->history_.erase(pos);
+  this->erase(pos);
   return true;
 }
 
+// ----------------------------------------------------------------------------
+bool
+track
+::remove( frame_id_t frame )
+{
+  auto const iter = this->find( frame );
+  if( iter == this->end() )
+  {
+    return false;
+  }
+  this->erase( iter );
+  return true;
+}
+
+// ----------------------------------------------------------------------------
+track::history_const_itr
+track
+::erase( history_const_itr iter )
+{
+  if( *iter )
+  {
+    ( *iter )->track_.reset();
+  }
+#if defined(__GNUC__) && !defined(__clang__) && __GNUC__ < 4
```

Comment:
This looks okay to me, but I wondering if we should have, or maybe already do have, some way of marking code sections that are special cases non-compliant C++11 compilers.  This way when we decide to stop supporting gcc 4 we can easily find which code can be cleaned up.  This could be a standard KWIVER macro that defines what you have here for some standard TODO style comment.  Maybe @dstoup or @linus-sherrill know if we've done something like that elsewhere.

Reply:
(Well, it _might_ be okay if I'd written `<= 4` or `< 5` 😳...)

I'm not aware of any such mechanism offhand... and on that note, I found without too much difficulty places supporting as far back as at least GCC 4.4, which I'm pretty sure we don't support. In any case, there are plenty of places checking compiler versions, and it probably wouldn't hurt to audit them at some point.

BTW, strictly speaking, the C++11 conforming code does work on GCC 4.9, but at this point (especially when we're already thinking about dropping GCC < 7) I think that's obscure enough that I didn't feel like making the effort to support it explicitly. The work-around code is *correct*, just less performant.